### PR TITLE
Use ibig instead of num_bigint

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by chriseth** [comment]
I think this does indeed provide a 20 to 50% speedup.

**Comment by chriseth** [comment]
ibig on top of the evaulation cache does not improve the fixed column generation much, but it does improve our evaluation benchmark, so I think it is still good to have.

File: number/Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ csv = "1.3"
 serde = { version = "1.0", default-features = false, features = ["alloc", "derive", "rc"] }
 serde_with = "3.6.1"
 schemars = { version = "0.8.16", features = ["preserve_order"]}
+ibig = "0.3.6"
```
Comment (leonardoalt) [inline number/Cargo.toml:22]: is `num-bigint` above still needed?

File: number/src/macros.rs [hunk 2]
```diff
@@ -70,25 +70,25 @@ macro_rules! powdr_field {
             }
         }
 
-        impl Shl<u64> for LargeIntImpl {
+        impl Shl<usize> for LargeIntImpl {
             type Output = Self;
 
-            fn shl(self, other: u64) -> Self {
+            fn shl(self, other: usize) -> Self {
                 // TODO: avoid using BigUint
```
Comment (leonardoalt) [inline number/src/macros.rs:77]: Remove this comment (and same comment below) and import?

File: number/Cargo.toml (hunk 1 above)
Comment (chriseth) [reply number/Cargo.toml:22]: Yes, because the large integer type of ark can only convert from num_bigint.

**Review by leonardoalt — APPROVED** [review]
please squash

**Pull request merged**